src/server/main_service.cc

```diff
@@ -2201,6 +2216,18 @@ void Service::Pubsub(CmdArgList args, ConnectionContext* cntx) {
     PubsubChannels(pattern, cntx);
   } else if (subcmd == "NUMPAT") {
     PubsubPatterns(cntx);
+  } else if (subcmd == "NUMSUB") {
+    if (args.size() == 1) {
+      (*cntx)->SendEmptyArray();
+      return;
+    }
+
+    int channels_size = args.size() - 1;
+    std::vector<string_view> channels(channels_size);
+    for (auto i = 0; i < channels_size; i++) {
+      channels[i] = ArgS(args, i + 1);
+    }
```

Comment:
You could just pass `args` to `PubsubNumSub()` instead of building a temporary `vector` just in order to build another `vector`

Reply:
fixed: https://github.com/dragonflydb/dragonfly/pull/2282/commits/4078b3426e136329e17ac3d80eea8baf931ffb45